velox/exec/tests/TableScanTest.cpp

```diff
@@ -2476,6 +2476,130 @@ TEST_F(TableScanTest, groupedExecutionWithOutputBuffer) {
       std::unordered_set<int32_t>({1, 5, 8}), getCompletedSplitGroups(task));
 }
 
+// Here we test various aspects of grouped/bucketed execution involving
+// output buffer and 3 pipelines.
+TEST_F(TableScanTest, groupedExecutionWithHashJoin) {
+  // Create source file - we will read from it in 6 splits.
+  auto vectors = makeVectors(10, 100);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->path, vectors);
+
+  // A chain of three pipelines separated by local exchange with the leaf one
+  // having scan running grouped execution - this will make all three pipelines
+  // running grouped execution.
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  core::PlanNodeId probeScanNodeId;
+  core::PlanNodeId buildScanNodeId;
+
+  auto pipe0Node =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(rowType_)
+          .capturePlanNodeId(probeScanNodeId)
+          .project({"c3 as x", "c2 as y", "c1 as z", "c0 as w", "c4", "c5"})
+          .hashJoin(
```

Comment:
Should we move this test to HashJoinTest? And also add a similar test to CrossJoinTest?

CC: @xiaoxmeng

Reply:
@mbasmanova 
We have other grouped execution tests here. It is not about joins but grouped execution.
Should we move all grouped execution tests to a separate cpp?
Also, I believe 'exec' tests executable is huge, should we start splitting it into multiple ones?

Reply:
> Should we move all grouped execution tests to a separate cpp?

That might be helpful. 

>  'exec' tests executable is huge, should we start splitting it into multiple ones?

Might be helpful as well.